GVRf/Framework/jni/objects/components/texture_capturer.cpp

```diff
@@ -164,8 +167,10 @@ glm::mat4 TextureCapturer::getMvpMatrix(float half_width, float half_height) {
 }
 
 void TextureCapturer::callback(int msg, char *info) {
-    Java_org_gearvrf_NativeTextureCapturer_callbackFromNative(
-            mJNIEnv, mCapturerObject, msg, info);
+    if (nullptr != mCapturerObject) {
```

Comment:
Feel free to make the proper fix in a subsequent commit. After all you know this code better. Also please fix the circular reference - the native side adds a reference to the java object. The java object cannot be ever collected because of that reference. Without collecting the java object the native can't be deleted either. Using the listeners is a kludge I had to invent to remove the leaks. There are no guarantees users will remove their listeners - if they don't, they will leak also whatever their listeners refer to.